feat: selectable students waiting step

johnvente opened the pull request:
### Description
This PR adds a toggable feature that allows instructors to automatically search a student after selecting it from the ORA waiting list table:

![ora-update-demo](https://github.com/openedx/edx-ora2/assets/134975835/4f84def3-7fe2-4d24-9b6f-ac13da97f662)



Here's what you see in the video explained:
- There's a student in the ORA waiting list after not receiving an assessment from their peers
- The instructor selects the Open Responses tab from the instructor dashboard
- The table shows the selectable column
- The instructor selects the number of students waiting for an assessment
- The instructor selects the student from the table
- A `Search` button appears
- The instructor presses the search button
- The information on the student's submission is shown
Now the instructor can grade the student.

Why are we proposing these changes? This improvement makes it easier for the instructors to grade ORA submissions without copying and pasting each username for a manual search. As said before, this feature can be turned on and off in case you want the default ORA behavior.

### How to test

- Create at least two students.
- Go to your course -> Create a new unit -> Create an open response component -> `Peer Assessment`.
- Then edit the open response -> Edit -> Assessment Steps.
      -> Select Step Peer Assessment option.
           `Must Grade`: 1
          `Grade By`: 1
- As students upload the open responses:
- Have student 1 fill out the open response and submit.
- Have student 2 fill out the open response, submit, and check student 1's open response.
- As an `instructor`, go to the Instructor panel ->` Open Responses `-> `Waiting` column.
- As an `instructor`, you can see the list of students. Select the one that you want to review and click on the `Search Learner` button.
     

#### DETAILED INSTRUCTIONS, WITH CONFIGURATIONS #####
You must enable these two features
```
FEATURES["ENABLE_XBLOCK_VIEW_ENDPOINT"] = True
FEATURES["ENABLE_ORA_SELECTABLE_LEARNER_WAITING_REVIEW"] = True
```

Here an example to add them in tutor:

```
from tutor import hooks

hooks.Filters.ENV_PATCHES.add_items(
    [
        (
            "openedx-lms-development-settings",
            """
FEATURES["ENABLE_XBLOCK_VIEW_ENDPOINT"] = True
FEATURES["ENABLE_ORA_SELECTABLE_LEARNER_WAITING_REVIEW"] = True

"""
        ),
    ]
)

```

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- [x] Translations and JS/SASS compiled
- [x] Bumped version number in [openassessment/\_\_init\_\_.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) and [package.json](https://github.com/openedx/edx-ora2/blob/master/package.json#L3)

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

openedx-webhooks commented:
Thanks for the pull request, @johnvente!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

<!-- comment:end_of_wip -->




<!-- comment:external_pr -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IHRydWV9 -->

mariajgrimaldi commented on openassessment/conf/locale/es_419/LC_MESSAGES/djangojs.po:
```diff
@@ -558,3 +558,11 @@ msgstr "Este ejercio de ORA ya ha sido liberado. Los cambios solo afectarán a l
+
+#: xblock/static/dist/openassessment-lms.018c55279d5e3b52e8ea.js:375
+msgid "You must select one row"
+msgstr "Debes seleccionar una fila"
```
missing end of line

mariajgrimaldi commented on openassessment/conf/locale/es_ES/LC_MESSAGES/djangojs.po:
```diff
@@ -557,3 +557,12 @@ msgstr "Esta tarea de respuesta abierta ya ha sido publicada. Los cambios solo a
+
+#: xblock/static/dist/openassessment-lms.018c55279d5e3b52e8ea.js:375
+msgid "You must select one row"
+msgstr "Debes seleccionar una fila"
```
same comment ^

mariajgrimaldi commented on openassessment/xblock/openassessmentblock.py:
```diff
@@ -648,6 +653,7 @@ def waiting_step_details_view(self, context=None):  # pylint: disable=unused-arg
         context_dict = {
             "title": self.title,
             "peer_assessment_required": peer_assessment_required,
+            "selectable_learners": ENABLE_SELECTABLE_LEARNER_WAITING_REVIEW.is_enabled()
```
```suggestion
            "selectable_learners_enabled": ENABLE_SELECTABLE_LEARNER_WAITING_REVIEW.is_enabled()
```

mariajgrimaldi commented on openassessment/xblock/static/dist/manifest.json:
```diff
@@ -1,22 +1,22 @@
-  "openassessment-rtl.js.map": "/openassessment-rtl.e023263ad9da2305253a.js.map",
-  "openassessment-studio.js": "/openassessment-studio.13c817d25360969539ff.js",
-  "openassessment-studio.js.map": "/openassessment-studio.13c817d25360969539ff.js.map",
+  "openassessment-editor-textarea.js": "/openassessment-editor-textarea.2e8cdb1f6b6447472326.js",
```
[curious] Why are these changes needed?

mariajgrimaldi commented on openassessment/xblock/static/js/spec/lms/components/WaitingStepList.spec.jsx:
```diff
@@ -0,0 +1,205 @@
+describe('OpenAssessment.WaitingStepList', () => {
+  window.gettext = sinon.fake((text) => text);;
+
+  describe('With enabled feature', () => {
```
can we be more specific with these descriptions?

mariajgrimaldi commented on openassessment/xblock/static/js/spec/lms/components/WaitingStepList.spec.jsx:
```diff
@@ -0,0 +1,205 @@
+        </IntlProvider>
+      );
+
+      // Find the DataTable component using its role
```
Are these in-line comments needed? We usually use in-line comments to explain why we're doing something that's not obvious when reading the code -- not the case here I think

mariajgrimaldi replied:
Maybe some are needed, but not all ;)

mariajgrimaldi commented on openassessment/xblock/static/js/spec/lms/components/WaitingStepList.spec.jsx:
```diff
@@ -0,0 +1,205 @@
+  window.gettext = sinon.fake((text) => text);;
+
+  describe('With enabled feature', () => {
+    it('should pass a basic test', () => {
+      expect(true).toBe(true);
+    });
```
is this test useful?

mariajgrimaldi commented on openassessment/xblock/static/js/spec/lms/components/WaitingStepList.spec.jsx:
```diff
@@ -0,0 +1,205 @@
+      expect(true).toBe(true);
+    });
+
+    it('allows row selection when ENABLE_MY_FEATURE is true', async () => {
```
can we use the real feature name?

mariajgrimaldi commented on openassessment/xblock/static/js/src/lms/components/WaitingStepContent.jsx:
```diff
@@ -55,17 +59,21 @@ const WaitingStepContent = ({ waitingStepDetails, refreshData }) => {
     student_data: PropTypes.arrayOf(PropTypes.object),
   }).isRequired,
   refreshData: PropTypes.func,
+  findUsername: PropTypes.func,
```
what about `findStudent`?

mariajgrimaldi commented on openassessment/xblock/static/js/src/lms/components/WaitingStepContent.jsx:
```diff
@@ -55,17 +59,21 @@ const WaitingStepContent = ({ waitingStepDetails, refreshData }) => {
   }).isRequired,
   refreshData: PropTypes.func,
+  findUsername: PropTypes.func,
+  selectableLearners: PropTypes.bool,
```
```suggestion
  selectableLearnersEnabled: PropTypes.bool,
```

mariajgrimaldi commented on openassessment/xblock/static/js/src/lms/components/WaitingStepList.jsx:
```diff
@@ -1,12 +1,31 @@
-  const studentListWithTimeAgo = studentList.map((item) => ({
+const currentDate = new Date();
+
+// Calculate the date seven days ago
+const sevenDaysAgo = new Date(currentDate);
+sevenDaysAgo.setDate(currentDate.getDate() - 7);
+
+const myObject = {
+  username: 'sofie_20',
+  graded: 1,
+  graded_by: 1,
+  created_at: sevenDaysAgo,
+  staff_grade_status: 'waiting',
+  workflow_status: '',
+};
+
```
what do we need to do to use existing users? This is pretty handy for testing though, at least for now.

mariajgrimaldi commented on openassessment/xblock/static/js/src/lms/components/WaitingStepList.jsx:
```diff
@@ -1,12 +1,31 @@
+const WaitingStepList = ({
+  studentList, refreshData, findUsername, selectableLearners,
+}) => {
+  const finalList = studentList.length ? studentList : [myObject];
```
is this final list temporary, right?

johnvente replied:
All this code is temporary, it's to test it faster. it will be rmoved

mariajgrimaldi commented on openassessment/xblock/static/js/src/lms/components/WaitingStepList.jsx:
```diff
@@ -1,12 +1,31 @@
+  studentList, refreshData, findUsername, selectableLearners,
+}) => {
+  const finalList = studentList.length ? studentList : [myObject];
+  const studentListWithTimeAgo = finalList.map((item) => ({
```
what's the function of this list?

johnvente replied:
It calculates how much time have happend for example have this date:  2023-08-04
returns 5 days ago

mariajgrimaldi commented on openassessment/xblock/static/js/src/lms/components/WaitingStepList.jsx:
```diff
@@ -15,10 +34,53 @@ const WaitingStepList = ({ studentList, refreshData }) => {
     <Button onClick={() => refreshData()}>{gettext('Refresh')}</Button>
   );
 
+  const FindLearnerAction = (props) => {
```
What about:
```suggestion
  const FindStudentAction = (props) => {
```

mariajgrimaldi commented on openassessment/xblock/static/js/src/lms/components/WaitingStepList.jsx:
```diff
@@ -15,10 +34,53 @@ const WaitingStepList = ({ studentList, refreshData }) => {
   );
 
+  const FindLearnerAction = (props) => {
+    const { tableInstance: { selectedFlatRows = [] } } = props || { tableInstance: {} };
```
I'm not sure what's selectedFlatRows or tableInstance. Can we use a better names?

johnvente replied:
I can't change the name of those, are names that come from [DataTable paragon](https://paragon-openedx.netlify.app/components/datatable/) component

mariajgrimaldi commented on openassessment/xblock/static/js/src/lms/components/WaitingStepList.jsx:
```diff
@@ -15,10 +34,53 @@ const WaitingStepList = ({ studentList, refreshData }) => {
+  const FindLearnerAction = (props) => {
+    const { tableInstance: { selectedFlatRows = [] } } = props || { tableInstance: {} };
+
+    const selectedFlatRowsLength = selectedFlatRows.length;
+
+    if (selectableLearners && selectedFlatRowsLength) {
```
```suggestion
    if (selectableLearners && selectedFlatRows.length) {
```

johnvente replied:
That constant is used [here](https://github.com/openedx/edx-ora2/pull/2025/files#diff-422421bc53f8bd0e9e53aee9916ae3184ef580902fa8fb5b40cad5f5befb0266R48) as well

mariajgrimaldi commented on openassessment/xblock/static/js/src/lms/components/WaitingStepList.jsx:
```diff
@@ -15,10 +34,53 @@ const WaitingStepList = ({ studentList, refreshData }) => {
+    const selectedFlatRowsLength = selectedFlatRows.length;
+
+    if (selectableLearners && selectedFlatRowsLength) {
+      const [row] = selectedFlatRows;
```
Same comment here with the naming, I think we can do better than `row`

johnvente replied:
Any idea? I can use rowStudent or rowTable

mariajgrimaldi commented on openassessment/xblock/static/js/src/lms/components/WaitingStepList.jsx:
```diff
@@ -15,10 +34,53 @@ const WaitingStepList = ({ studentList, refreshData }) => {
+    if (selectableLearners && selectedFlatRowsLength) {
+      const [row] = selectedFlatRows;
+      const {
+        values: { username },
```
`student_username` perhaps?

mariajgrimaldi commented on openassessment/xblock/static/js/src/lms/components/WaitingStepList.jsx:
```diff
@@ -15,10 +34,53 @@ const WaitingStepList = ({ studentList, refreshData }) => {
+
+      const handleFindLearnerClick = () => {
+        if (findUsername) {
+          findUsername(username);
```
what does it do `findUsername`? finds the username? The name is a bit misleading.

mariajgrimaldi commented on openassessment/xblock/static/js/src/lms/components/WaitingStepList.jsx:
```diff
@@ -15,10 +34,53 @@ const WaitingStepList = ({ studentList, refreshData }) => {
+            disabled={invalidSelection}
+            onClick={handleFindLearnerClick}
+          >
+            {gettext('Search learner')}
```
I tested the translation and it didn't work:
![image](https://github.com/openedx/edx-ora2/assets/64440265/5f7dd932-37db-4797-bfbc-17692715e5aa)
Maybe we're missing something. I'll research a bit but I'm not sure I'll find something since I'm not familiar with this repo. We can ask for help later when this PR is ready for review!

mariajgrimaldi commented on openassessment/xblock/static/js/src/lms/containers/WaitingStepDetailsContainer.jsx:
```diff
@@ -20,15 +20,45 @@ const WaitingStepDetailsContainer = ({ waitingStepDataUrl, onMount }) => {
+    }
+  };
+
+  const getUserNamSelected = (username) => {
```
```suggestion
  const getUsernameSelected = (username) => {
```

mariajgrimaldi commented on openassessment/xblock/static/js/src/lms/containers/WaitingStepDetailsContainer.jsx:
```diff
@@ -20,15 +20,45 @@ const WaitingStepDetailsContainer = ({ waitingStepDataUrl, onMount }) => {
+  };
+
+  const getUserNamSelected = (username) => {
+    const button = document.querySelector('.button-staff-tools');
```
is querySelectos the best way of getting the user's info?

mariajgrimaldi replied:
What if the classes or attributes are changed?

johnvente replied:
We would have to re do the component that carry the data of the student, remember that this has django template mixed and make other request

johnvente replied:
I don't think so that can be changed, those classes are  in css code and js code

mariajgrimaldi replied:
Is this pattern of using querySelector used elsewhere in this repo?

johnvente replied:
The React part is something new, other parts are using Jquery that is not an option

mariajgrimaldi commented on openassessment/xblock/static/js/src/lms/components/WaitingStepContent.jsx:
```diff
@@ -55,17 +59,21 @@ const WaitingStepContent = ({ waitingStepDetails, refreshData }) => {
 WaitingStepContent.defaultProps = {
   refreshData: () => ({}),
+  findUsername: undefined,
+  selectableLearners: undefined,
```
why not False?

mariajgrimaldi commented on openassessment/xblock/static/js/src/lms/components/WaitingStepContent.jsx:
```diff
@@ -55,17 +59,21 @@ const WaitingStepContent = ({ waitingStepDetails, refreshData }) => {
 
 WaitingStepContent.defaultProps = {
   refreshData: () => ({}),
+  findUsername: undefined,
```
why not returning ({}) as refreshData?

mariajgrimaldi reviewed (changes requested):
I left a few comments for you to consider! Thanks

johnvente commented:
> I left a few comments for you to consider! Thanks

Hey @mariajgrimaldi! I've done the changes, any additional suggestion I will be attentive to review it

mariajgrimaldi commented on openassessment/xblock/static/js/src/lms/components/WaitingStepContent.jsx:
```diff
@@ -3,7 +3,9 @@ import PropTypes from 'prop-types';
 
-const WaitingStepContent = ({ waitingStepDetails, refreshData }) => {
+const WaitingStepContent = ({
+  waitingStepDetails, refreshData, findStudent, selectableLearnersEnabled,
```
We should be consistent about our naming conventions. If we're using students then we should use student throughout the implementation:
```suggestion
  waitingStepDetails, refreshData, findStudent, selectableStudentsEnabled,
```

mariajgrimaldi replied:
Thus, we should use `student` instead of learners throughout the implementation. In this repo, do they call them learners or students?

johnvente replied:
I'll call them student

mariajgrimaldi commented on openassessment/xblock/static/js/src/lms/components/WaitingStepContent.jsx:
```diff
@@ -55,17 +59,21 @@ const WaitingStepContent = ({ waitingStepDetails, refreshData }) => {
-    must_grade: PropTypes.string,
-    waiting_count: PropTypes.string,
-    overwritten_count: PropTypes.string,
+    must_be_graded_by: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    must_grade: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    waiting_count: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    overwritten_count: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
```
[courious] why change this from PropTypes.string to PropTypes.onOfType?

johnvente replied:
There is an error here, it watis an number but something recieves an string idk why that is happening

mariajgrimaldi commented on openassessment/xblock/static/js/spec/lms/components/WaitingStepList.spec.jsx:
```diff
@@ -0,0 +1,181 @@
+
+      const studentList = [
+        {
+          username: 'jhon_20',
```
We should use dummy names for testing

mariajgrimaldi commented:
Hi there: @openedx/content-aurora 

This PR is an improvement on the current students' waiting list grading workflow by instructors, which can be turned on/off.

We've encountered this ORA improvement as part of the on-campus learning experience effort (Project with the Spanish Consortium). Specifically, we were requested to:

> The rows of students waiting for peer assessments table must be selectable so instructors can select the student to display grading/relevant information without copy/pasting the students' username

This is our first implementation approach for the improvement! We're open to suggestions. Please, let us know what's the next step for this to land :)

mariajgrimaldi commented:
Hello there, @itsjeyd! Thanks for the quick response! There isn't a ticket for this specific improvement, so creating one is better. As [I mentioned before](https://github.com/openedx/edx-ora2/pull/2025#issuecomment-1677949381), this effort is part of the On Campus initiative, which might be helpful to include in the ticket. Let me know what else you need from us!

e0d commented:
@Daniel-hershel is this PR made irrelevant by the work toward the ORA grading MFE?

Daniel-hershel commented:
@e0d I don't have a deep understanding of this PR, but after looking into it a little bit I don't think we are doing anything that addresses this specific issue so I believe this is a valid PR. I don't see any conflicts from a product perspective. @mattcarter can confirm here, or we can officially move this to an eng review potentially,

leangseu-edx commented:
What would happen when you select multiple students?

johnvente commented:
> What would happen when you select multiple students?

It appears a message that says "You must select one row"

leangseu-edx commented:
> > What would happen when you select multiple students?
> 
> @leangseu-edx It appears a message that says "You must select one row"

If this use case is for single row action, it is probably better to do action column like https://paragon-openedx.netlify.app/components/datatable/#actions

itsjeyd commented:
Hi @Daniel-hershel, based on your most recent comment (https://github.com/openedx/edx-ora2/pull/2025#issuecomment-1697540941), can we consider this PR approved from product perspective?

CC @mattcarter

johnvente commented:
Hey @leangseu-edx! I've changed this as you mentioned [here](https://github.com/openedx/edx-ora2/pull/2025#issuecomment-1697901284) as action row the gift was updated as well. I've added the translations for two words "Action" and "Review"
Review is not working for me, do you know what might it be? Thanks!

itsjeyd commented:
Hi @Daniel-hershel and @mattcarter, a friendly ping about a question I had raised [earlier](https://github.com/openedx/edx-ora2/pull/2025#issuecomment-1717526028) -- can we consider this PR approved from product perspective?

Daniel-hershel commented:
Yes this is approved from the product perspective.

itsjeyd commented:
@Daniel-hershel Great, thanks!

@mattcarter @leangseu-edx Tests are green, too, so we should be good to go for engineering review now.

CC @openedx/content-aurora

e0d commented:
@Daniel-hershel it would be great to have a rough sense of review timing as this is blocking need for the Spanish Consortium.

mattcarter reviewed (approved):
approved!

johnvente commented:
> Hey @leangseu-edx! I've changed this as you mentioned [here](https://github.com/openedx/edx-ora2/pull/2025#issuecomment-1697901284) as action row the gif was updated as well. I've added the translations for two words "Action" and "Review" "Review" is not working for me, do you know what might it be? Thanks!

Hi @mattcarter! Thanks a lot for approving this. I have a little blocker with this comment, do know what it might be?

johnvente commented:
> approved!
Hey @mattcarter! We've decided to follow up this to get a merge here so what it would be the next step to merge this? We will cover the translation in another issue

leangseu-edx commented:
> @Daniel-hershel Great, thanks!
> 
> @mattcarter @leangseu-edx Tests are green, too, so we should be good to go for engineering review now.
> 
> CC @openedx/content-aurora

Sorry for late response. Since you already implemented the action button, what would be the of checkbox now? Those button can always just be visible and checkbox can go away. You can only click one action at a time. Unless checkbox is serving different purpose.

mattcarter commented:
> > approved!
> 
> Hey @mattcarter! We've decided to follow up this to get a merge here so what it would be the next step to merge this? We will cover the translation in another issue

Hi @johnvente - good question!  There is a notable feature being built in a feature branch. We're checking to see what impact it has on your PR.

itsjeyd commented:
@mattcarter @Daniel-hershel Just to understand the status of this PR better, will you still want to complete a round of product review before merging it?

johnvente commented:
> > > approved!
> > 
> > 
> > Hey @mattcarter! We've decided to follow up this to get a merge here so what it would be the next step to merge this? We will cover the translation in another issue
> 
> Hi @johnvente - good question! There is a notable feature being built in a feature branch. We're checking to see what impact it has on your PR.



> > > approved!
> > 
> > 
> > Hey @mattcarter! We've decided to follow up this to get a merge here so what it would be the next step to merge this? We will cover the translation in another issue
> 
> Hi @johnvente - good question! There is a notable feature being built in a feature branch. We're checking to see what impact it has on your PR.



> > > approved!
> > 
> > 
> > Hey @mattcarter! We've decided to follow up this to get a merge here so what it would be the next step to merge this? We will cover the translation in another issue
> 
> Hi @johnvente - good question! There is a notable feature being built in a feature branch. We're checking to see what impact it has on your PR.

Hi @mattcarter thanks for letting me know. I've seen that I have some conflicts that I need to resolve once I resolved them what would we need to mergering?

itsjeyd commented:
Hey @mattcarter, a friendly reminder to have a look at @johnvente's question above.

johnvente commented:
> Hi @mattcarter thanks for letting me know. I've seen that I have some conflicts that I need to resolve once I resolved them what would we need to mergering?

Hi @mattcarter! Could you help me with previous question over here? Sorry, but I would like to merge this ASAP

johnvente commented:
Hey @mattcarter! Any update over here?

jalondonot commented:
Hi @e0d and @mphilbrick211. We're struggling to get a final review and approval to merge this PR. Do you have any ideas or suggestions to unblock the process? Thanks!

mphilbrick211 commented:
Hi @jalondonot! I'll look into this for you.

e0d commented:
@johnvente will try to get the finalized ASAP, can you address the conflicts?

johnvente commented:
@e0d Hi there! Sorry for the delay it's okay now. Thanks a lot!

johnvente commented:
> @johnvente It looks like there are some conflicts left to resolve. Could you please have another look?
> 
> CC @e0d

Hi there @itsjeyd! I've solved the conflicts

johnvente commented:
I will come back here when the workflows are being running well after that I will solve these new conflicts
commit with problem(https://github.com/openedx/edx-ora2/commit/68fc523ef5335c2a7124f373e73b27b8750b2f93)

e0d commented:
@johnvente the checks have all passed.

johnvente commented:
> @johnvente the checks have all passed.

Hi @e0d! yep, but if I do a rebase they will fail so it would be better to rebase when the last merges have the workflows fine. Then I will check the conflicts

itsjeyd commented:
Hey @johnvente!

> ... if I do a rebase they will fail so it would be better to rebase when the last merges have the workflows fine. Then I will check the conflicts

Do I understand correctly that this PR has become blocked on other work for now? And if so could you list the PRs that you're waiting on here for reference?

johnvente commented:
> Do I understand correctly that this PR has become blocked on other work for now? And if so could you list the PRs that you're waiting on here for reference?

Hi there @itsjeyd for now to this repository is only this one. I have another pr that are made for others repositories would you like that I share them? I'm waiting for a merge with the workflows fine

itsjeyd commented:
@johnvente 

> I have others PR's that are made for others repositories would you like that I share them?

Only if they are related to (or blocking) this PR :)

> I'm waiting for a merge when the workflows are fine

Are you saying that there is a problem on the `master` branch that causes checks/workflows to fail? If so, it would be great if you could open an [issue](https://github.com/openedx/edx-ora2/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc) that describes the problem. That way, it might not take as long for someone to create a PR with a fix.

itsjeyd commented:
Hey @johnvente, a friendly nudge about my comment above.

johnvente commented:
> Hey @johnvente, a friendly nudge about my comment above.

Hi there @itsjeyd! Sorry for the delay. Since this PR has been blocked I didn't see necessary to create an issue for the workflows, now they are fine so I will solve the conflicts now

johnvente commented:
Hi @itsjeyd conflicts were solved now. I will be checking the commits here and reduce them but now its okay

johnvente commented:
Hi @itsjeyd I've solve some conflicts and squashed the commits, this PR is ready to merge

itsjeyd commented:
@johnvente Great to see that this is no longer blocked on build issues! And thank you for working on resolving conflicts and squashing commits.

As of today it looks like there are some additional conflicts to solve.

@mariajgrimaldi When do you think you'd be able to re-review this PR? I'm asking because if we could schedule the review for a specific day, @johnvente could plan to resolve the conflicts right before you look at the code again (instead of having to do it multiple times while waiting for review).

Let us know if that would work for you? :)

mariajgrimaldi reviewed (approved):
@itsjeyd: all my comments were addressed, so this looks good to me. Thank you folks!

johnvente commented:
Hi there @itsjeyd! conflicts solved

mariajgrimaldi commented:
Thanks @itsjeyd! And no, I don't have write access here.

pomegranited reviewed (changes requested):
Thank you for submitting this feature @johnvente ! I've reviewed the code and tested it in my browser as you describe on the PR description, and it works exactly as expected. 

Could you do a version bump too please? Needed in two places, cf [example](https://github.com/openedx/edx-ora2/pull/2173/commits/b1cd0d616235da8c1982ca6ebc42cb0beaa5612e).

Will approve and merge once that and my comment about the feature flag docs are addressed.

a11y note: Since DataTable is used for the Waiting list, it's nicely accessible. However the table showing the "full list" of ORA responses is _not_ accessible, so I was not able to use my keyboard to navigate to the "waiting" link. Fixing this is out of scope for this PR, but I wanted to flag it in case anyone is looking at this.

johnvente commented:
Hi there @pomegranited! Thanks for taking a look over here. I will check the bump version and the docs for the feature flag ASAP

johnvente commented:
Hi @pomegranited! I've added the doc for the feature flag and updated the version. Please let me know if there is any changed needed it. Thanks a lot

pomegranited commented on openassessment/xblock/openassessmentblock.py:
```diff
@@ -69,6 +70,10 @@
 
 logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
 
+ENABLE_SELECTABLE_LEARNER_WAITING_REVIEW = SettingDictToggle(
+    "FEATURES", "ENABLE_SELECTABLE_LEARNER_WAITING_REVIEW", default=False, module_name=__name__
+)
+
```
Thank you for documenting this flag @johnvente ! Just going to wait a bit to see what @feanil has to say.

pomegranited replied:
Hmm.. haven't heard anything back, but I think the code tells us the answer.

@johnvente have you seen how the other feature flags are used by ORA2?

cf
https://github.com/openedx/edx-ora2/blob/7c33111d1cffad523a5123477a9fcbcfe545ba07/settings/base.py#L148
https://github.com/openedx/edx-ora2/blob/7c33111d1cffad523a5123477a9fcbcfe545ba07/openassessment/xblock/config_mixin.py#L19

I think we should follow that convention and move this flag + docs into an `ENABLE_ORA_SELECTABLE_LEARNER_WAITING_REVIEW` flag in lms/cms common ([e.g. like ENABLE_ORA_ALL_FILE_URLS in [lms.common.env](https://github.com/openedx/edx-platform/blob/2f2ed4d6cbfbd42e2d7590e5bfef45f31b33bd55/lms/envs/common.py#L885-L897) and [cms.common.env](https://github.com/openedx/edx-platform/blob/2f2ed4d6cbfbd42e2d7590e5bfef45f31b33bd55/cms/envs/common.py#L397-L409)), and use the ORA xblock.config_mixin to check the flag itself.

pomegranited replied:
Hi @johnvente , sorry for my delays here.

I've [nudged Feanil on Slack](https://openedx.slack.com/archives/C02RFB0K6GH/p1708635211633609), but if he doesn't get back to us, then I'll finish my review and merge this when I start my week on Tuesday.

CC @itsjeyd

johnvente replied:
HI @pomegranited! sorry for the delay, I've changed the feature flag as you mentioned now is called `ENABLE_ORA_SELECTABLE_LEARNER_WAITING_REVIEW` I've updated the PR's description as well

itsjeyd commented:
Hey @pomegranited, just checking in to see when you'd be able to complete another review pass here?

pomegranited commented on openassessment/conf/locale/es_ES/LC_MESSAGES/djangojs.po:
```diff
@@ -589,3 +589,13 @@ msgstr "Esta tarea de respuesta abierta ya ha sido publicada. Los cambios solo a
+#: openassessment/xblock/static/dist/openassessment-studio.979e8b88dd0d9cee68f7.js:25
+#: openassessment/xblock/static/js/src/lms/components/WaitingStepList.jsx:38
+msgid "Action"
+msgstr "Acción"
```
[OEP-58](https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0058-arch-translations-management.html) is a work in progress, but I don't think it's enough to put these translations here.. I think they need to be set in Transifex, otherwise the next transifex pull will override them.

If this happens, could you work with the #translations-wg to resolve it?

(Not a blocker for merge here, just FYI.)

pomegranited reviewed (approved):
:+1: This change works as described, thank you @johnvente :)

Could you merge latest `master`, fix the conflicts, and bump the version strings again? I can merge once that's done.

- [x] I tested this on my devstack using the PR test instructions (which were fantastic 💯 )
- [x] I read through the code
- [x] I checked for accessibility issues by using my keyboard to navigate.
- [x] Includes documentation of the added feature flag.
- [x] User-facing strings are extracted for translation

johnvente commented:
> 👍 This change works as described, thank you @johnvente :)
> 
> Could you merge latest `master`, fix the conflicts, and bump the version strings again? I can merge once that's done.
> 

Hi @pomegranited Thanks for the review here we really appreciate it. I've solve conflicts and the comment that you mentioned [here](https://github.com/openedx/edx-ora2/pull/2025#discussion_r1488704595) Now this is ready to be merged if there is any concern please let me know. Thanks a lot

johnvente commented on openassessment/xblock/test/test_openassessment.py:
```diff
@@ -308,6 +308,31 @@ def test_ora_blocks_listing_view_include_esg_flag(self, xblock, esg_flag_input,
+        xblock_arg_path = "//script[contains(@type, 'json/xblock-args')]"
+
+        xblock_args_el = tree.xpath(xblock_arg_path)
+        json.loads(xblock_args_el[0].text)['CONTEXT']['selectable_learners_enabled'] = esg_flag_input
```
@pomegranited Done!

openedx-webhooks commented:
@johnvente 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-ora2&entry.867055334=https://github.com/openedx/edx-ora2/pull/2025&entry.1484655318=https://github.com/johnvente&entry.752974735=2023-08-09+14:54&entry.1917517419=2024-03-01+21:50&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
